test/integration/policy_info_metric_test.go

```diff
@@ -0,0 +1,149 @@
+			if err != nil {
+				return err
+			}
+			return len(routeList.Items)
+		}, defaultTimeoutSeconds, 1).Should(Or(Equal(0), Equal(1)))
+
+		if len(routeList.Items) == 0 {
+			By("Exposing the metrics service as a route")
+			_, err = oc("expose", "service", metricsSvc.Name, "-n", ocmNS, `--overrides={"spec":{"tls":{"termination":"reencrypt"}}}`)
+			Expect(err).To(BeNil())
+
+			Eventually(func() interface{} {
+				var err error
+				routeList, err = clientHubDynamic.Resource(common.GvrRoute).Namespace(ocmNS).List(context.TODO(), metav1.ListOptions{LabelSelector: propagatorMetricsSelector})
+				if err != nil {
+					return err
+				}
+				return len(routeList.Items)
+			}, defaultTimeoutSeconds, 1).Should(Equal(1))
+		}
+
+		routeURL = routeList.Items[0].Object["spec"].(map[string]interface{})["host"].(string)
+		By("Got the metrics route url: " + routeURL)
+	})
+	It("Checks that the endpoint does not expose metrics without auth", func() {
+		Eventually(func() interface{} {
+			resp, err := curlMetricsRoute()
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds, 1).Should(ContainSubstring("Unauthorized"))
+	})
+	It("Checks that endpoint has a HELP comment for the metric", func() {
+		token, err := oc("whoami", "-t")
+		Expect(err).To(BeNil())
+		Eventually(func() interface{} {
+			resp, err := curlMetricsRoute("--header", "Authorization: Bearer "+token)
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds, 1).Should(ContainSubstring("HELP " + metricName))
+	})
+	It("Checks that a compliant policy reports a metric of 0", func() {
+		By("Creating a compliant policy")
+		oc("apply", "-f", compliantPolicyYaml, "-n", userNamespace, "--kubeconfig="+kubeconfigHub)
+		Eventually(
+			getComplianceState(compliantPolicyName),
+			defaultTimeoutSeconds,
+			1,
+		).Should(Equal(policiesv1.Compliant))
+
+		By("Checking the policy metric")
+		token, err := oc("whoami", "-t")
+		Expect(err).To(BeNil())
+		regex := `(?m)` + metricName + `{.*policy="` + compliantPolicyName + `.*} 0$`
+		Eventually(func() interface{} {
+			resp, err := curlMetricsRoute("--header", "Authorization: Bearer "+token)
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds, 1).Should(MatchRegexp(regex))
+	})
+	It("Checks that a noncompliant policy reports a metric of 1", func() {
+		By("Creating a noncompliant policy")
+		oc("apply", "-f", noncompliantPolicyYaml, "-n", userNamespace, "--kubeconfig="+kubeconfigHub)
+		Eventually(
+			getComplianceState(noncompliantPolicyName),
+			defaultTimeoutSeconds,
+			1,
+		).Should(Equal(policiesv1.NonCompliant))
+
+		By("Checking the policy metric")
+		token, err := oc("whoami", "-t")
+		Expect(err).To(BeNil())
+		regex := `(?m)` + metricName + `{.*policy="` + noncompliantPolicyName + `.*} 1$`
```

Comment:
Since this is so similar to the previous regex, it might be nice to have a function that determines if it's a violation:
```go
func isPolicyNonCompliant(policyName, metricsOutput string) (bool, error) {
	regex := regexp.MustCompile(`(?m)` + metricName + `{.*policy="` + policyName + `".*} (?P<violation>\d+)$`)
	matches := regex.FindStringSubmatch(metricsOutput)
	if len(matches) == 0 {
		return false, fmt.Errorf("the policy %s was not present in the metrics output", policyName)
	}

	return matches[regex.SubexpIndex("violation")] == "1", nil
}
```

Reply:
Good point